pilot/pkg/bootstrap/server.go

```diff
@@ -927,14 +927,19 @@ func (s *Server) initIstiodCerts(args *PilotArgs, host string) error {
 
 // maybeInitDNSCerts initializes DNS certs if needed.
 func (s *Server) maybeInitDNSCerts(args *PilotArgs, host string) error {
-	// Generate DNS certificates only if custom certs are not provided via args.
-	if !hasCustomTLSCerts(args.ServerOptions.TLSOptions) && s.EnableCA() {
-		// Create DNS certificates. This allows injector, validation to work without Citadel, and
+	if hasCustomTLSCerts(args.ServerOptions.TLSOptions) {
+		// Use the DNS certificate provided via args.
+		// This allows injector, validation to work without Citadel, and
 		// allows secure SDS connections to Istiod.
-		log.Infof("initializing Istiod DNS certificates host: %s, custom host: %s", host, features.IstiodServiceCustomHost.Get())
-		if err := s.initDNSCerts(host, features.IstiodServiceCustomHost.Get(), args.Namespace); err != nil {
-			return err
-		}
+		return nil
+	}
+	if !s.EnableCA() && features.PilotCertProvider.Get() == IstiodCAProvider {
+		// If CA functionality is disabled, istiod cannot sign the DNS certificates.
+		return fmt.Errorf("cannot create self-signed DNS certificate when Istiod CA is disabled")
```

Comment:
should this be an error? do we actually require dns certs to exist?

Reply:
I think Istiod should fail starting if there's no DNS cert. We should make it an requirement for Istiod to always have a secure port. I don't see uses case that this can't be achieved - they should at least use plugin DNS cert or Citadel, if K8s CA is not available.

Reply:
we could probably return nil for now - it doesn't hurt for Istiod to just exist in the cluster without a DNS certificate. It won't be useful for anything, but probably doesn't make sense to return an error

Reply:
Don't have a strong opinion on this, will defer to Oliver's comment